service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -301,8 +304,16 @@ public ResponseEntity<MethodListResponse> getMethods(
 
     List<MethodDetails> methodDetails;
     if (methodVersionId != null) {
-      methodDetails =
-          List.of(methodVersionToMethodDetails(methodVersionDao.getMethodVersion(methodVersionId)));
+      try {
+        methodDetails =
+            List.of(
+                methodVersionToMethodDetails(methodVersionDao.getMethodVersion(methodVersionId)));
+      } catch (bio.terra.cbas.common.exceptions.MethodProcessingException
+          | java.net.MalformedURLException
+          | java.net.URISyntaxException
+          | bio.terra.dockstore.client.ApiException e) {
+        return ResponseEntity.badRequest().body(new MethodListResponse().error(e.toString()));
```

Comment:
reviewers, is this the right thing to do when the method version URL can't be resolved?

Reply:
Should this be Internal Server Error? If it was a user issue / bad request then the method wouldn't have been imported only no? 🤔

Reply:
you're right, bad request seemed wrong - internal server error sounds good to me